src/plugins/data_views/common/data_views/data_views.ts

```diff
@@ -397,13 +397,11 @@ export class DataViewsService {
 
   /**
    * Clear index pattern list cache.
-   * @param id optionally clear a single id
+   * @param full clears all caches
    */
-  clearCache = (id?: string) => {
+  clearCache = (full?: boolean) => {
     this.savedObjectsCache = null;
-    if (id) {
-      this.dataViewCache.clear(id);
-    } else {
+    if (full) {
```

Comment:
It seems this code had some undesirable functional overlap to begin with. the `savedObjectsCache` is really a list of data views whereas the `dataViewCache` actually stores DataView instances. They'd be cleared in slightly different situations, whether you're getting a fresh list or fresh DataView instance. What are the needs and intentions of the api consumer code? I think this gets lost when `clearCache` always clears the data view list cache and optionally clears the dataView cache. Determining api consumers intentions might be more work than the code changes in which case perhaps we should deprecate this method and replace it with two new methods.